### ISSUE TITLE
Session set_flashdata() - Error '502 Bad Gateway'

### ISSUE THREAD

**Issue by julianfuerderer** [bcit-ci/CodeIgniter#2476]
Hello,

i dont know if it is a bug:

I got an 502 Bad Gateway-Error after an Ajax-Request.
In the target controller/method i have setted 3 flashdata-session-items:

$this->session->set_flashdata("success","true");
$this->session->set_flashdata("success_to_user_id","1");
$this->session->set_flashdata("success_something_else","xyz");

The request successfully finished. But then, when i refresh the site
(window.location.reload or F5) i got an '502 Bad Gateway' Error.

After playing around I have fixed that error:

I reduced the 3 set_flashdata() calls to 1. Then it works, no more 502.

So:

Less or equal 2x $this->session->set_flashdata(); = OK, NO ERROR

Greater or equal 3x $this->session->set_flashdata(); = ERROR 502 BAD GATEWAY

Is this a bug ? 

Greetings,
Julian Fuerderer

**Comment by narfbg** [comment]
Fixed by #3073.

**Comment by Easter1021** [comment]
@AdenFraser thank you.

in Nginx, you must set conf with:

```
    location ~ \.php$ {
    	try_files $uri =404;
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass php:9000;
        fastcgi_index index.php;
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param PATH_INFO $fastcgi_path_info;
        fastcgi_buffer_size 128k;
        fastcgi_buffers 4 256k;
        fastcgi_busy_buffers_size 256k;
        fastcgi_temp_file_write_size 256k;
        fastcgi_intercept_errors on;
    }
```
